### DISCUSSION TITLE
Feature: restrict server listen to Scrypted Server Addresses

### DISCUSSION THREAD

**Discussion by rmaz** [koush/scrypted#1452] [Ideas]
At the moment the scrypted node server will perform a `listen()` with no address specified:
https://github.com/koush/scrypted/blob/6d520dc4b24f2f20378ab7f7cbfd5d1d57fe6510/server/src/scrypted-server-main.ts#L45

On a multi network system this will listen on all interfaces. When hosting on OpenWRT or similar style routers, this will have the unexpected effect of opening HTTP ports on the WAN port as we are using the host network when launching the docker container.

It could be a simple change to use the user provided list of interfaces from `scrypted.addressSettings.getLocalAddresses()` instead to restrict the server listen to the LAN interface, which seems desirable.

**Comment by koush** [comment]
I wouldn't put Scrypted on your router or give it direct WAN access. Lots more ports are opened besides this one.
